devolving_music/lib/song_scores.py

```diff
@@ -1,176 +1,191 @@
-
-        # sorts by peakyness
-        return song_submissions_scored
-
-    @staticmethod
-    def mvg_avg(song_submissions_sorted: Iterable["SongSubmission"]):
-
+    def get_compare_submission_random(self, submission_id) -> int:
+        key_list = list(self.song_score_dict.keys())
+        key_list.remove(submission_id)
+        # once you have a critical number of comparisons then pull from quality
+        # list
+        return random.choice(key_list)
+
+    def get_compare_submission_linear(self, submission_id) -> int:
+        if(len(self.comparison_submissions) == 0):
+            return self.get_compare_submission_random(submission_id)
+        first_recent_id = self.comparison_submissions[-1].first_submission.id
+        if(submission_id == first_recent_id):
+            return self.get_compare_submission_random(submission_id)
+        return first_recent_id
+
+    def mvg_avg(
+            self,
+            song_submissions_sorted: Iterable["ScoreSuite"]) -> Iterable[int]:
         mvg_avg = [None] * len(song_submissions_sorted)
 
         # song_submissions_scored is a list of song submissions
-
         # that have been sorted into prepeak,peak,and post peak
-
         # these prepeak, peak, and post posteak have each been sorted
         # accordingly with energy
-
         # that have been scored with current comparisons
-
         # get a moving average corresponding to quality score
 
         return mvg_avg
 
-    @staticmethod
-    def check_quality(
-            song_submissions_sorted: Iterable["SongSubmission"],
-            remove: int):
+    def check_quality(self,
+                      song_submissions_sorted: Iterable["ScoreSuite"],
+                      remove: int) -> Iterable["ScoreSuite"]:
 
         song_submissions_pruned = song_submissions_sorted
 
         # do (song_submissions_sorted[i].quality_score - mvg_avg[i])
-
         # for every song in score submission
-
         # return n indices with smallest values
-
         # these indices are to be removed
-        return song_submissions_pruned
 
-    @staticmethod
-    def get_final_list(comparison_submissions: Iterable["SongComparison"],
+        return song_submissions_pruned
 
+    def get_dict_from_keys(self, new_keys) -> "dict[int, ScoreSuite]":
+        return {key: self.song_score_dict[key] for key in new_keys}
 
-                       song_submissions: Iterable["SongSubmission"]):
+    def get_final_list(self) -> Iterable["ScoreSuite"]:
 
-        scored_submissions = SongSubmission.get_scores(
-            comparison_submissions, song_submissions)
+        scored_submissions = list(self.get_scores().values())
 
         # remove all song_submissions with no information
+        info_list = SongScores.get_info_sort(scored_submissions)
 
-        # sort by peakyness
-
-        # peaky_sorted_submissions=get_peak_sort(scored_submissions)
+        informed_list = list(filter(lambda sub: sub.info_score > 0, info_list))
```

Comment:
```suggestion
        informed_list = list(filter(lambda sub: sub.info_score >= INFO_THRESHOLD, info_list))
```

Let's pull this out into a global constant so it's not a magic number. Notice that I changed it to `>=` so if you accept this suggestion, `INFO_THRESHOLD` should be 1 to keep parity with what you have here.

Reply:
fair enough